backend/arm64/emit.ml

```diff
@@ -2068,47 +2083,59 @@ let fundecl fundecl =
   assert (List.length !call_gc_sites = num_call_gc);
   (match fun_end_label with
   | None -> ()
-  | Some fun_end_label -> emit_printf "%a:\n" femit_label fun_end_label);
-  cfi_endproc ();
-  emit_symbol_type femit_symbol fundecl.fun_name "function";
-  emit_symbol_size fundecl.fun_name;
+  | Some fun_end_label ->
+    let fun_end_label = label_to_asm_label ~section:Text fun_end_label in
+    D.define_label fun_end_label);
+  D.cfi_endproc ();
+  (* The type symbol and the size are system specific. They are not output on
+     macOS. The asm directives take care of correctly handling this distinction.
+     For the size, they automatically emit the size [. - symbol], meaning "this
+     minus symbol definition". *)
+  D.type_symbol ~ty:Function fun_sym;
+  D.size fun_sym;
   emit_literals ()
 
 (* Emission of data *)
 
 let emit_item (d : Cmm.data_item) =
```

Comment:
This function should be able to be shared with amd64/emit.ml and moved to emitaux.ml; let's leave a CR.